spec/features/course/achievement_management_spec.rb

```diff
@@ -7,15 +7,17 @@
     let(:course) { create(:course) }
     let(:other_achievement) { create(:course_achievement, course: course) }
     let(:achievement_condition) do
-      create(:course_condition_achievement,
-             course: course,
-             achievement: other_achievement)
+      create(:achievement_condition, course: course, achievement: other_achievement)
+    end
+    let(:assessment) { create(:assessment, course: course) }
+    let(:assessment_condition) do
+      create(:assessment_condition, course: course, assessment: assessment)
     end
-    let(:level_condition) { create(:course_condition_level, course: course) }
-    let(:achievement) do
+    let(:level_condition) { create(:level_condition, course: course) }
+    let!(:achievement) do
```

Comment:
especially an eagerly-definined variable. We had a few nasty things happen with this.